File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/viewmodel/ReservaViewModel.java
package com.example.agnciadeturismo.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.agnciadeturismo.data.repository.ReservaRepositoryTask;
import com.example.agnciadeturismo.model.ReservaDto;

import java.util.ArrayList;

public class ReservaViewModel extends ViewModel {

    private MutableLiveData<Boolean> mCadastrado = new MutableLiveData<>();
    private MutableLiveData<Integer> mCodigo = new MutableLiveData<>();
    private MutableLiveData<ArrayList<ReservaDto>> mAllReservas = new MutableLiveData<>();

    public LiveData reservas = mAllReservas;
    public LiveData cadastrado = mCadastrado;
    public LiveData codigoReserva = mCodigo;

    public void cadastrarReserva(int codigoCartao, String cpf, double valorTotal, int status, String data){
        new Thread(new Runnable() {
            @Override
            public void run() {
                mCadastrado.postValue(ReservaRepositoryTask.cadastrarReserva(codigoCartao, cpf, valorTotal, status, data));
            }
        }).start();
    }

    public void getCodigoReserva(int codigoCartao, String cpf, double valorTotal, int status, String data){
        new Thread(new Runnable() {
            @Override
            public void run() {
                mCodigo.postValue(ReservaRepositoryTask.getCodigoReserva(codigoCartao, cpf, valorTotal, status, data));
            }
        }).start();
    }

    public void getAllReserva(String cpf){
        new Thread(new Runnable() {
            @Override
            public void run() {
                mAllReservas.postValue(ReservaRepositoryTask.getAllReserva(cpf));
            }
        }).start();
    }
}

File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/presenter/viewmodel/ReservaViewModel.kt
package com.example.agnciadeturismo.presenter.viewmodel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.agnciadeturismo.data.repository.ReservaRepositoryTask
import com.example.agnciadeturismo.model.ReservaDto
import java.lang.Exception
import java.util.*

class ReservaViewModel(private val reservaRepositoryTask: ReservaRepositoryTask) : ViewModel() {

    private val TAG = "ReservaViewModel"
    private val mCadastrado = MutableLiveData<Boolean>()
    private val mCodigo = MutableLiveData<Int>()
    private val mAllReservas = MutableLiveData<ArrayList<ReservaDto>>()

    var reservas: LiveData<ArrayList<ReservaDto>> = mAllReservas
    var cadastrado: LiveData<Boolean> = mCadastrado
    var codigoReserva: LiveData<Int> = mCodigo

    fun cadastrarReserva(codigoCartao: Int, cpf: String?, valorTotal: Double, status: Int, data: String?) {
        Thread {
            try {
                mCadastrado.postValue(
                    reservaRepositoryTask.cadastrarReserva(
                        codigoCartao,
                        cpf,
                        valorTotal,
                        status,
                        data
                    )
                )
            }catch (ex : Exception){Log.d(TAG, ex.toString())}

        }.start()
    }

    fun getCodigoReserva(codigoCartao: Int, cpf: String?, valorTotal: Double, status: Int, data: String?) {
        Thread {
            try {
                mCodigo.postValue(
                    reservaRepositoryTask.getCodigoReserva(
                        codigoCartao,
                        cpf,
                        valorTotal,
                        status,
                        data
                    )
                )
            }catch (ex : Exception){Log.d(TAG, ex.toString())}
        }.start()
    }

    fun getAllReserva(cpf: String?) {
        Thread {
            try {
                mAllReservas.postValue(reservaRepositoryTask.getAllReserva(cpf))
            }catch (ex : Exception){Log.d(TAG, ex.toString())}
        }.start()
    }

    class ViewModelFactory(private val reservaRepositoryTask: ReservaRepositoryTask) : ViewModelProvider.Factory {
        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
            return modelClass.getConstructor(ReservaRepositoryTask::class.java)
                .newInstance(reservaRepositoryTask)
        }
    }

}